qiita_db/data.py

```diff
@@ -1257,6 +1257,60 @@ def create(cls, processed_params_table, processed_params_id, filepaths,
         pd.add_filepaths(filepaths, conn_handler)
         return cls(pd_id)
 
+    @classmethod
+    def delete(cls, processed_data_id):
+        """Removes the processed data with id processed_data_id
+
+        Parameters
+        ----------
+        processed_data_id : int
+            The processed data id
+
+        Raises
+        ------
+        QiitaDBUnknownIDError
```

Comment:
You are not raising this error, so no need to list here. In fact, there is no call to `exists` so probably useful to add.
Also, you're missing the description of the `QiitaDBError`.